### DISCUSSION TITLE
Support `formAction` on `select` and other interactive form elements

### DISCUSSION THREAD

**Discussion by IstoraMandiri** [vercel/next.js#53826] [Ideas]
### Goals

Currently the [docs](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions#formaction) state that `formAction` is supported for a select number form elements: 

> button, input type="submit", and input type="image"

This is good for many use cases, but shouldn't this also be supported for all interactive form elements, such as `select`, `checkbox`, `radio`, etc?

### Background

I am finding `formAction` to be useful for rapid prototyping in Nextjs apps that use `experimental.serverActions`. 

Of course, there are workarounds to do this through client side for proper form controls, or by rendering a series of buttons, but it feels like the following pattern should be supported in a server component:

```typescript
<select formAction={serverAction}>
  {ROLES.map((role) => (
    <option
      key={role}
      value={role}
      selected={role === member.role}
    >
      {role}
    </option>
  ))}
</select>
```

### Proposal

Review all the possible native [form elements](https://www.w3schools.com/html/html_form_elements.asp) and for those with interactivity, support `formAction`, passing in the current element's data to the action a la `input type="image"`.

**Comment by robertwbradford** [comment]
I am not sure how to trigger a `formAction` directly from the `<select>`, but I have a similar use case where one of my form controls is outside of the associated `<form>` element and I'm using the `form` attribute to associate it. For example, it have the following server action:

```js
async function myServerAction(formData) {
  "use server";

  // Do some things with the form data...
}
```

I then have the following search form:

```jsx
<form action={myServerAction} id="searchform">
  <label>
    Search
    <input type="text" name="query" />
  </label>
  <button type="submit" aria-label="Search">🔍</button>
</form>
```

Hitting enter while in the `query` input or clicking the submit button successfully runs the `myServerAction`. The form data also includes a `sort` value from an associated form control that exists elsewhere on the page by the results:

```jsx
<label>
  Sort
  <select
    name="sort"
    form="searchform" // associate the form
    value={useSearchParams().get("sort")}
    onChange={(event) => event.target.form.requestSubmit()}
  >
    <option value="recent">Recent</option>
    <option value="name">Name</option>
  </select>
</label>
```
For its `onChange` originally I had `event.target.form.submit()` but it was erring and React prompted me to use [`requestSubmit`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/requestSubmit) instead.
